tests/test_treasury.cairo

```diff
@@ -0,0 +1,292 @@
+    gov_contract_address.serialize(ref calldata);
+    AMM_contract_address.serialize(ref calldata);
+
+    // Precalculate the address to obtain the contract address before the constructor call (deploy) itself
+    let contract_address = contract.precalculate_address(@calldata);
+
+    prank(CheatTarget::One(contract_address), gov_contract_address, CheatSpan::TargetCalls(1));
+    let deployed_contract = contract.deploy(@calldata).unwrap();
+    (deployed_contract, contract_address)
+}
+
+fn get_important_addresses() -> (
+    ContractAddress, ContractAddress, ContractAddress, ContractAddress
+) {
+    let gov_contract_address: ContractAddress = testStorage::gov_contract_address
+        .try_into()
+        .unwrap();
+    let AMM_contract_address: ContractAddress = testStorage::AMM_contract_address
+        .try_into()
+        .unwrap();
+    let (treasury_contract_address, pre_calculated_address) = deploy_contract(
+        "Treasury", gov_contract_address, AMM_contract_address
+    );
+    return (
+        gov_contract_address,
+        AMM_contract_address,
+        treasury_contract_address,
+        pre_calculated_address
+    );
+}
+
+#[test]
+fn test_contract_deployment() {
+    let (
+        _gov_contract_address,
+        _AMM_contract_address,
+        treasury_contract_address,
+        pre_calculated_address
+    ) =
+        get_important_addresses();
+    assert(treasury_contract_address == pre_calculated_address, 'Deployment Error');
+}
+
+#[test]
+#[fork("GOERLI")]
+fn test_send_tokens_to_address() {
+    let (gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let user1: ContractAddress = 0x026fa92011b2f27eca57a44411837e38a4313dfb11d561146039b445815db35b
+        .try_into()
+        .unwrap();
+    let user2: ContractAddress = '0xUser2'.try_into().unwrap();
+    let token: ContractAddress = 0x049D36570D4e46f48e99674bd3fcc84644DdD6b96F7C741B1562B82f9e004dC7
+        .try_into()
+        .unwrap();
+    let decimal: u256 = 1_000000000000000000;
+
+    prank(CheatTarget::One(token), user1, CheatSpan::TargetCalls(1));
+    IERC20Dispatcher { contract_address: token }.transfer(treasury_contract_address, 1 * decimal);
+
+    let user2_bal_before_transfer = IERC20Dispatcher { contract_address: token }.balanceOf(user2);
+
+    prank(
+        CheatTarget::One(treasury_contract_address), gov_contract_address, CheatSpan::TargetCalls(1)
+    );
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .send_tokens_to_address(user2, 1 * decimal, token);
+
+    let user2_bal_after_transfer = IERC20Dispatcher { contract_address: token }.balanceOf(user2);
+
+    assert(user2_bal_before_transfer != user2_bal_after_transfer, 'token transfer Error');
+    assert(user2_bal_after_transfer == 1 * decimal, 'Transfer calculation error');
+}
+
+#[test]
+#[should_panic(expected: ('Caller is not the owner',))]
+#[fork("GOERLI")]
```

Comment:
Goerli will be shut down in April and is already deprecated. Use Sepolia

Reply:
I'll correct that also. Could you please send the Carmine Juno rpc url for sepolia